File: 01/js/merge.js
const TestHelper = require('./testHelper.js');
const Algorithm = require('./algorithm.js');

const n = 50000;
// const arr = TestHelper.generateRandomArray(n, 0, n);
const arr = TestHelper.generateNearlyOrderedArray(n, 10);
const arr2 = TestHelper.copyArray(arr);
const arr3 = TestHelper.copyArray(arr);

TestHelper.testSort('        merge sort', Algorithm.mergeSort, arr);
TestHelper.testSort('      2 merge sort', Algorithm.mergeSort2, arr2);
TestHelper.testSort('insertionSort sort', Algorithm.insertionSort, arr3);

File: 01/cpp/merge.cpp
#include <iostream>
#include "testHelper.h"
#include "algorithm.h"

using namespace std;

int main() {
	int n = 50000;
	// int *arr = TestHelper::generateRandomArray(n, 0, n);
	int *arr = TestHelper::generateNearlyOrderedArray(n, 10);
	int *arr2 = TestHelper::copyIntArray(arr, n);
	int *arr3 = TestHelper::copyIntArray(arr, n);

	TestHelper::testSort("        merge sort", Algorithm::mergeSort, arr, n);
	TestHelper::testSort("      2 merge sort", Algorithm::mergeSort2, arr2, n);
	TestHelper::testSort("insertionSort sort", Algorithm::insertionSort, arr3, n);

	delete[] arr;
	delete[] arr2;
	delete[] arr3;

	return 0;
}